File: analyzer-direct/src/main/scala/io/github/tassiLuca/analyzer/client/AnalyzerView.scala
package io.github.tassiLuca.analyzer.client

trait AnalyzerView:
  def run(): Unit
  def update(result: OrganizationReport): Unit
  def endComputation(): Unit
  def error(errorMessage: String): Unit

object AnalyzerView:
  def gui(controller: AppController): AnalyzerView = AnalyzerGUI(controller)

File: analyzer-direct-kt/src/main/kotlin/io/github/tassiLuca/analyzer/client/AnalyzerView.kt
package io.github.tassiLuca.analyzer.client

import io.github.tassiLuca.analyzer.lib.RepositoryReport

typealias OrganizationReport = Pair<Map<String, Long>, Set<RepositoryReport>>

/** A view for the analyzer application. */
interface AnalyzerView {

    /** Runs the view. */
    fun run()

    /** Updates the view with the given [result]. */
    fun update(result: OrganizationReport)

    /** Notify the view the end of the computation. */
    fun endComputation()

    /** Display the given [errorMessage]. */
    fun error(errorMessage: String)

    /** Notify the view the computation has been cancelled. */
    fun cancelled()
}
